### PR TITLE
update gitignore and analysis_options

### PR DESCRIPTION
**Pull request by devoncarew**
Update gitignore and analysis_options in the generated app.

### CODE REVIEW

**Comment by Hixie** [comment]
rslgtm

**Comment by devoncarew** [comment]
Sorry, put this out for review prematurely. Updated to:

- have put `ios --init` and `create` create the user-modifiable resources (`create` now creates an ios/ directory)
- update the .gitignore file
- `create` also creates an `android/res/` directory, so we have one less warning at apk build time

PTAL

**Comment by devoncarew** [comment]
fixes https://github.com/flutter/flutter/issues/1562

**Comment by devoncarew** [comment]
also fix https://github.com/flutter/flutter/pull/1802

File: packages/flutter_tools/lib/src/commands/create.dart [hunk 1]
```diff
@@ -145,8 +148,16 @@ class FlutterSimpleTemplate extends Template {
...
     files['pubspec.yaml'] = _pubspec;
     files['README.md'] = _readme;
-    files['android/AndroidManifest.xml'] = _apkManifest;
     files['lib/main.dart'] = _libMain;
+
+    // Android files.
+    files['android/AndroidManifest.xml'] = _apkManifest;
+    // Create a file here, so we create the directory for the user (and it gets
+    // committed with git).
+    files['android/res/readme.md'] = _androidResReadme;
```
Comment (eseidelGoogle) [inline packages/flutter_tools/lib/src/commands/create.dart]: Do you mean README.md?

Comment (devoncarew) [inline packages/flutter_tools/lib/src/commands/create.dart]: Hmm, sort of. I can change this to match the root readme name. I've just never been a fan of all caps file names.

Comment (devoncarew) [inline packages/flutter_tools/lib/src/commands/create.dart]: Changed to `README.md`.

**Comment by Hixie** [comment]
LGTM

**Pull request merged**